test/functional/tests/volumes/test_many_cores_on_many_lvms.py

```diff
@@ -0,0 +1,117 @@
+          - FIO with verification ran successfully.
+          - Configuration after reboot match configuration before.
+    """
+    with TestRun.step(f"Prepare devices."):
+
+        cache_dev, core_dev = prepare_devices()
+
+    with TestRun.step("Create LVMs."):
+        lvm_filters = []
+        pv_num = 1
+        vg_num = 1
+        lv_num = 16
+        cache_num = 1
+        cas_dev_num = 16
+
+        config = LvmConfiguration(lvm_filters, pv_num, vg_num, lv_num, cache_num, cas_dev_num)
+
+        lvms = Lvm.create_specific_lvm_configuration(core_dev, config, lvm_as_core=True)
+
+    with TestRun.step(f"Create CAS device."):
+        cache = casadm.start_cache(cache_dev[0], force=True)
+        cores = []
+        for lvm in lvms:
+            cores.append(cache.add_core(lvm))
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(cores)
+
+    with TestRun.step("Flush buffers"):
+        for core in cores:
+            TestRun.executor.run(f"hdparm -f {core.path}")
+
+    with TestRun.step("Create init config from running configuration"):
+        InitConfig.create_init_config_from_running_configuration()
+        config_before_reboot = fs_utils.read_file(opencas_conf_path)
+        devices_before = get_block_devices_list()
+
+    with TestRun.step("Reboot system."):
+        TestRun.executor.reboot()
+
+    with TestRun.step("Validate running configuration"):
+        InitConfig.create_init_config_from_running_configuration()
+        config_after_reboot = fs_utils.read_file(opencas_conf_path)
+        devices_after = get_block_devices_list()
+
+        if config_after_reboot == config_before_reboot and devices_after == devices_before:
+            TestRun.LOGGER.info(f"Configuration is as expected")
+        else:
+            TestRun.LOGGER.error(f"Configuration changed after reboot.")
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(cores)
+
+    with TestRun.step("Remove CAS devices."):
+        casadm.remove_all_detached_cores()
+        casadm.stop_all_caches()
+
+    with TestRun.step("Remove LVMs."):
+        Lvm.remove_all()
+
+
+def prepare_devices():
+    cache_device = TestRun.disks['cache']
+    core_device = TestRun.disks['core1']
+
+    cache_device.create_partitions([Size(10, Unit.GibiByte)])
+    core_device.create_partitions([Size(10, Unit.GibiByte)])
+
+    return cache_device.partitions, core_device.partitions
+
+
+def run_fio_on_lvm(volumes):
+    fio_run = (Fio().create_command()
+               .read_write(ReadWrite.randrw)
+               .io_engine(IoEngine.sync)
+               .io_depth(1)
+               .time_based()
+               .run_time(datetime.timedelta(seconds=180))
+               .do_verify()
+               .verify(VerifyMethod.md5)
+               .block_size(Size(1, Unit.Blocks4096)))
+
+    for volume in volumes:
+        fio_run.add_job().target(volume).size(volume.size)
+    fio_run.run()
```

Comment:
I see these functions in your few tests from the same category. Maybe extract them to another file - 'lvm_common_utils' or something like that. Also my comments to `prepare_devices` from other PRs are applicable here too.